### PR TITLE
deps: remove personal forks/branches, and add a note about why

### PR DESCRIPTION
**Pull request by benesch**
Maintainership burden can't be shared with personal forks, and personal branches tend to disappear and make old versions of Materialize uncompilable.

### Motivation

   * This PR refactors existing code.

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [x] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  - n/a

### CODE REVIEW

**Review by umanwizard — CHANGES_REQUESTED** [review]
In practice what would have happened had this policy existed yesterday is that Gábor would have abandoned [his PR](https://github.com/MaterializeInc/materialize/pull/19329) (which he already did, see [here](https://materializeinc.slack.com/archives/CM7ATT65S/p1684327908814659?thread_ts=1684318187.564809&cid=CM7ATT65S)), I wouldn't have helped him get it into a mergeable state, and Materialize would simply not have been buildable with gcc 13 until someone with "create repo" permissions got around to fixing it.

Requiring people to ask in a channel and wait around for one of a very small set of people to respond is _way_ too much friction for fixing a dubiously-useful lint error. I really don't think all three policies "no duplicate packages allowed", "no personal forks allowed", and "unprivileged users can't create forks in the MaterializeInc org" can coexist.

**Comment by umanwizard** [comment]
To give context to the friction point: it took about an hour to fix this, from the point when I said in Slack that I'd take a look to the point when I uploaded the final commit that fixed it. Part of that was due to my own mistake (I was trying to switch to the main branch of Tonic rather than just cherry-picking a fix onto the branch we were using) but regardless, it's way higher than it should be and pretty close to the upper limit of what I'm willing to invest into helping fix this class of issue. If I had had to wait for approval, the amount of time it could have taken is unbounded -- sure I could have done something else in the meantime but context switching has its own cost.

**Review by umanwizard — APPROVED** [review]

**Comment by benesch** [comment]
@umanwizard and I talked offline and agreed that the best course of action is to lift the hard ban on duplicate dependencies. That will allow folks to temporarily introduce duplicate dependencies to get themselves unblocked. Forks will still need to be part of the MaterializeInc org, and will therefore continue to require requests in #eng-infra, but without the duplicate dependency ban we expect that we'll very rarely need to fork projects.

**Pull request merged**
